rest/route/project.go

```diff
@@ -346,77 +346,86 @@ func (h *projectIDPatchHandler) Run(ctx context.Context) gimlet.Responder {
-	dbProjectRef.Admins = []string{}
+	allAdmins := util.UniqueStrings(append(oldProject.Admins, newProjectRef.Admins...)) // get original and new admin
+	newProjectRef.Admins = []string{}
 	for _, admin := range allAdmins {
 		if !util.StringSliceContains(adminsToDelete, admin) {
-			dbProjectRef.Admins = append(dbProjectRef.Admins, admin)
+			newProjectRef.Admins = append(newProjectRef.Admins, admin)
 		}
 	}
 
 	// validate triggers before updating project
 	catcher := grip.NewSimpleCatcher()
-	for i, trigger := range dbProjectRef.Triggers {
-		catcher.Add(trigger.Validate(dbProjectRef.Identifier))
+	for i, trigger := range newProjectRef.Triggers {
+		catcher.Add(trigger.Validate(newProjectRef.Identifier))
 		if trigger.DefinitionID == "" {
-			dbProjectRef.Triggers[i].DefinitionID = util.RandomString()
+			newProjectRef.Triggers[i].DefinitionID = util.RandomString()
 		}
 	}
-	dbProjectRef.Triggers = append(p.Triggers, dbProjectRef.Triggers...)
+	newProjectRef.Triggers = append(oldProject.Triggers, newProjectRef.Triggers...)
 	if catcher.HasErrors() {
 		return gimlet.MakeJSONErrorResponder(errors.Wrap(catcher.Resolve(), "error validating triggers"))
 	}
 
-	newRevision := model.FromStringPtr(apiProjectRef.Revision)
+	newRevision := model.FromStringPtr(requestProjectRef.Revision)
 	if newRevision != "" {
 		if err = h.sc.UpdateProjectRevision(h.projectID, newRevision); err != nil {
 			return gimlet.MakeJSONErrorResponder(err)
 		}
-		dbProjectRef.RepotrackerError = &dbModel.RepositoryErrorDetails{
+		newProjectRef.RepotrackerError = &dbModel.RepositoryErrorDetails{
 			Exists:            false,
 			InvalidRevision:   "",
 			MergeBaseRevision: "",
 		}
 	}
 
 	// complete all updates
-	if err = h.sc.UpdateProject(dbProjectRef); err != nil {
+	if err = h.sc.UpdateProject(newProjectRef); err != nil {
 		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Database error for update() by project id '%s'", h.projectID))
 	}
-	if err = h.sc.UpdateProjectVars(h.projectID, &apiProjectRef.Variables, false); err != nil { // destructively modifies apiProjectRef.Variables
+	if err = h.sc.UpdateProjectVars(h.projectID, &requestProjectRef.Variables, false); err != nil { // destructively modifies requestProjectRef.Variables
 		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Database error updating variables for project '%s'", h.projectID))
 	}
-	if err = h.sc.UpdateProjectAliases(h.projectID, apiProjectRef.Aliases); err != nil {
+	if err = h.sc.UpdateProjectAliases(h.projectID, requestProjectRef.Aliases); err != nil {
 		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Database error updating aliases for project '%s'", h.projectID))
 	}
 
-	for i := range apiProjectRef.Subscriptions {
-		apiProjectRef.Subscriptions[i].OwnerType = model.ToStringPtr(string(event.OwnerTypeProject))
-		apiProjectRef.Subscriptions[i].Owner = model.ToStringPtr(h.projectID)
+	deleteAdmins := []string{}
+	for _, toDelete := range requestProjectRef.DeleteAdmins {
```

Comment:
Isn't this block the same as 356-358? Should be able to use adminsToDelete I think. Also I think this block makes more logical sense up there with the admin logic.

Reply:
Oops fixed. I actually want this to happen after the project is updated, in case that update fails